### PR TITLE
Feature/encrypt oauth tokens

### PR DESCRIPTION
**Pull request by baristaGeek**
## Description
We want to improve security and salt access tokens.

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)  
- [x] New feature (non-breaking change which adds functionality)  
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)  
- [ ] Documentation update  
- [ ] Chore: cleanup/renaming, etc  
- [ ] RFC  

## Acceptance
- [x] I have read [the Contributing guidelines](CONTRIBUTING.md)
- [x] I have read the [Code of Conduct](CODE_OF_CONDUCT.md)

### CODE REVIEW

File: utils/encryption/encrypt.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import base64ToUint8Array from "./base64ToUint8Array";
+
+export function encrypt(plaintext: string): string {
+    if(!process.env.ENCRYPTION_KEY) {
+        throw new Error("Encryption key not found"); 
+    }
+    const key = base64ToUint8Array(process.env.NEXT_PUBLIC_ENCRYPTION_KEY!);
```
Comment (EstebanDalelR) [inline utils/encryption/encrypt.ts:7]: Weird we have two things in the envelope

File: utils/encryption/decrypt.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import base64ToUint8Array from "./base64ToUint8Array";
+
+export default function decrypt(ciphertext: string): string {
```
Comment (EstebanDalelR) [inline utils/encryption/decrypt.ts:3]: I think the two processes are so coupled they could go in the same file.

File: utils/encryption/base64ToUint8Array.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+export default function base64ToUint8Array(base64String: string): Uint8Array {
+    const binaryString = atob(base64String);
+    const byteArray = new Uint8Array(binaryString.length);
```
Comment (EstebanDalelR) [inline utils/encryption/base64ToUint8Array.ts:3]: Same file is worth it.

File: utils/db/github/getToken.ts [hunk 4]
```diff
@@ -1,11 +1,12 @@
 import executeRequest from "../azuredb";
+import decrypt from "../../encryption/decrypt";
 
 export default async function getUser(user): Promise<any> {
   try {
     let data = await executeRequest(
       `EXEC dbo.get_github_token @watermelon_user = '${user}'`
     );
-    return data;
+    return decrypt(data);
```
Comment (EstebanDalelR) [inline utils/db/github/getToken.ts:9]: Is all data encrypted?

File: app/(loggedIn)/github/page.tsx [hunk 5]
```diff
@@ -53,13 +56,13 @@ export default async function ServicePage({
...
       headers: {
-        Authorization: `token ${json.access_token}`,
+        Authorization: `token ${decrypt(json.access_token)}`,
       },
     });
     let userJson = await user.json();
     // save user correctly
     await saveUserInfo({
-      access_token: json.access_token,
+      access_token: encrypt(json.access_token),
```
Comment (EstebanDalelR) [inline app/(loggedIn)/github/page.tsx:65]: Perf.

**Comment by baristaGeek** [comment]
Not a priority at this time

**Pull request closed without merging**